### PR TITLE
Update eslint rules and add husky pre commit hook

### PR DESCRIPTION
**Pull request by rohitkrai03**

### CODE REVIEW

File: src/components/ComponentSamples/ComponentSamplesPage.tsx [hunk 1]
```diff
@@ -108,7 +108,7 @@ export const ComponentSamplesPage = () => {
     if (formState.component) {
       setSelected(formState.component);
     }
-  }, []);
+  }, [formState.component]);
```
Comment (sahil143) [inline src/components/ComponentSamples/ComponentSamplesPage.tsx:111]: @rohitkrai03  can we remove this dependency. We just need setSelected called once when the component is mounted
```suggestion
  }, []);
```

File: src/pages/SamplePage/SamplePage.tsx [hunk 2]
```diff
@@ -1,11 +1,10 @@
...
-import { Link, withRouter } from 'react-router-dom';
-import { useDispatch } from 'react-redux';
-
 import { Button, StackItem, Stack, Title, Spinner } from '@patternfly/react-core';
+import { addNotification } from '@redhat-cloud-services/frontend-components-notifications/redux';
 import { Main } from '@redhat-cloud-services/frontend-components/Main';
 import { PageHeader, PageHeaderTitle } from '@redhat-cloud-services/frontend-components/PageHeader';
-import { addNotification } from '@redhat-cloud-services/frontend-components-notifications/redux';
+import { useDispatch } from 'react-redux';
+import { Link, withRouter } from 'react-router-dom';
```
Comment (sahil143) [inline src/pages/SamplePage/SamplePage.tsx:7]: not sure but shouldn't these come after 'react' import?

File: src/components/ComponentSamples/ComponentSamplesPage.tsx (hunk 1 above)
Comment (rohitkrai03) [reply src/components/ComponentSamples/ComponentSamplesPage.tsx]: Done. Added a comment and disabled the eslint rule.

File: src/pages/SamplePage/SamplePage.tsx (hunk 2 above)
Comment (rohitkrai03) [reply src/pages/SamplePage/SamplePage.tsx]: Added extra path groups to handle this scenario. By default `@` groups are sorted before normal imports.

**Review by sahil143 — APPROVED** [review]
/lgtm

**Pull request merged**